ci-operator/step-registry/cucushift/installer/rehearse/gcp/cases/valid-confidential-computing/provision/cucushift-installer-rehearse-gcp-cases-valid-confidential-computing-provision-commands.sh

```diff
@@ -0,0 +1,320 @@
+pull_secret=$(<"${CLUSTER_PROFILE_DIR}/pull-secret")
+
+patch=$(mktemp)
+expected_err_msg=""
+ret=0
+
+declare -A cc_type_machine_series_dict=(
+  [AMDEncryptedVirtualization]="c2d, n2d, c3d"
+  [AMDEncryptedVirtualizationNestedPaging]="n2d"
+  [IntelTrustedDomainExtensions]="c3"
+  [Enabled]="c2d, n2d, c3d"
+)
+
+declare -A instance_type_supported_cc_type_dict=(
+  [n2d-standard-4]="AMDEncryptedVirtualization AMDEncryptedVirtualizationNestedPaging Enabled"
+  [c2d-standard-4]="AMDEncryptedVirtualization Enabled"
+  [c3d-standard-4]="AMDEncryptedVirtualization Enabled"
+  [c3-standard-4]="IntelTrustedDomainExtensions"
+  [n2-standard-4]=""
+)
+
+declare -A instance_type_unsupported_cc_type_dict=(
+  [n2d-standard-4]="IntelTrustedDomainExtensions"
+  [c2d-standard-4]="IntelTrustedDomainExtensions AMDEncryptedVirtualizationNestedPaging"
+  [c3d-standard-4]="IntelTrustedDomainExtensions AMDEncryptedVirtualizationNestedPaging"
+  [c3-standard-4]="AMDEncryptedVirtualization AMDEncryptedVirtualizationNestedPaging Enabled"
+  [n2-standard-4]="AMDEncryptedVirtualization AMDEncryptedVirtualizationNestedPaging Enabled IntelTrustedDomainExtensions"
+)
+
+function save_artifacts()
+{
+  local -r install_dir="$1"
+  local -r testing_scenario_num="$2"
+  local current_time
+
+  set +o errexit
+  current_time=$(date +%s)
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${install_dir}/.openshift_install.log" > "${ARTIFACT_DIR}/cluster_${testing_scenario_num}_openshift_install-${current_time}.log"
+
+  set -o errexit
+}
+
+function create_install_config()
+{
+  local -r cluster_name=$1
+  local -r install_dir=$2
+
+  cat > ${install_dir}/install-config.yaml << EOF
+apiVersion: v1
+baseDomain: ${BASE_DOMAIN}
+compute:
+- architecture: amd64
+  hyperthreading: Enabled
+  name: worker
+  platform: {}
+  replicas: 3
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  platform: {}
+  replicas: 3
+metadata:
+  creationTimestamp: null
+  name: ${cluster_name}
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineNetwork:
+  - cidr: 10.0.0.0/16
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  gcp:
+    projectID: ${GOOGLE_PROJECT_ID}
+    region: ${REGION}
+publish: External
+pullSecret: >
+  ${pull_secret}
+sshKey: |
+  ${ssh_pub_key}
+EOF
+}
+
+function compose_expected_err_msg()
+{
+  local -r instance_type="$1"
+  local -r compute_cc_type="$2"
+  local -r control_plane_cc_type="$3"
+  local -r default_machine_platform_cc_type="$4"
+  local -r use_default_flag="$5"
+  local -r on_host_maintenance="$6"
+  local supported_cc_types unsupported_cc_types cc_type1 cc_type2
+
+  supported_cc_types=${instance_type_supported_cc_type_dict["${instance_type}"]}
+  unsupported_cc_types=${instance_type_unsupported_cc_type_dict["${instance_type}"]}
+
+  if ${use_default_flag}; then
```

Comment:
I think we can directly read install-config.yaml to get the expected value in a more flexible way like something in https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/installer/check/gcp/confidential-computing/cucushift-installer-check-gcp-confidential-computing-commands.sh#L34-L57, so that we can freely assemble install-config.yaml.

Reply:
FYI the function is called after "create manifests" command, so it's possible the install-config.yaml is absent.

Reply:
We can always save a copy of install-config.yaml before "create manifests"